### PR TITLE
DOC: use constants in performance-comparisons.ipynb

### PR DESCRIPTION
**Pull request by raybellwaves**
## Description

I've simplified the performance comparisons notebook by setting constants which can be adjusted at the top of each section e.g. `num_rows`. This makes it easier for anyone running this to adjust the value and hopefully not encounter memory values. It can also help with testing these benchmarks on dataframes of various lengths. I've stripped the output as I was working on a A10G and I couldn't run with the current `num_rows` value. I also didn't want to commit the results which may differ compared to the H100 which is used currently and I would rather the results be committed by the RAPIDS team. I can confirm the notebook runs end-to-end (you can see my version here: https://github.com/raybellwaves/cudf-performance-comparisons/blob/main/performance-comparisons.ipynb with smaller `num_rows` and smaller `timeit_number` on a A10G (EC2 machine))

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by bdice** [comment]
@raybellwaves Thanks for the PR! I skimmed it and the core ideas seem helpful.

@galipremsagar Would you be able to take a look at this and incorporate the proposed changes for next time we run benchmarks?

**Comment by galipremsagar** [comment]
I'll run these on H100's and update this PR

File: docs/cudf/source/user_guide/performance-comparisons/performance-comparisons.ipynb
Comment (bdice) [inline docs/cudf/source/user_guide/performance-comparisons/performance-comparisons.ipynb:1]: Can we fix (or hide) the warning here?
```
DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.
  lambda df: df.groupby(["key"], group_keys=False).apply(custom_formula_udf)
```

Also can we run on a system that isn't doing background work? I see some references to `tritonserver` consuming GPU memory (probably on other devices?) in the `nvidia-smi` output, which makes it hard to know if it affected performance negatively.

**Review by mroeschke — COMMENTED** [review]
Nice changes. When printing out the cudf version, could you also include the pandas version as well?

**Pull request merged**